test/request.js

```diff
@@ -275,45 +276,64 @@ describe('Request', () => {
 
     describe('active()', () => {
 
-        it('exits handler early when request is no longer active', async () => {
+        it('exits handler early when request is no longer active', { retry: true }, async (flags) => {
+
+            const onCleanup = [];
+            flags.onCleanup = async () => {
+
+                for (const cleanup of onCleanup) {
+                    await cleanup();
+                }
+            };
+
+            let testComplete = false;
+            onCleanup.unshift(() => {
+
+                testComplete = true;
```

Comment:
This needs to always run before other cleanups. Otherwise you risk that a previous cleanup never resolves, and `testComplete` would never be set, causing an infinite loop in the handler.

I would just put the logic directly into the `onCleanup` handler before the loop.